warehouse/query-core/src/main/java/datawave/query/common/grouping/GroupingUtil.java

```diff
@@ -177,15 +189,27 @@ private Multimap<String,String> getFieldToFieldWithGroupingContextMap(Document d
                     // Add GroupingTypeAttribute to fieldMap with a grouping context that is based on ordered attribute type
                     int i = 0;
                     for (Map.Entry<Type<?>,GroupingTypeAttribute<?>> sortedEntry : attrSortedMap.entries()) {
-                        String fieldNameWithContext = fieldName + "." + i++;
+                        String fieldNameWithContext = null;
+                        String context = convertGroupingContext(i++);
+                        if (containsContext) {
+                            // multivalued entries from separate entries, but the field names contain the same context number
+                            fieldNameWithContext = shortName + "." + context;
+                        } else {
+                            // multivalued entries that do not have grouping context
+                            fieldNameWithContext = fieldName + "." + context;
+                        }
+
                         fieldMap.put(fieldNameWithContext, sortedEntry.getValue());
                         fieldToFieldWithContextMap.put(shortName, fieldNameWithContext);
+                        contextSet.add(context);
                     }
                 } else {
                     GroupingTypeAttribute<?> created = new GroupingTypeAttribute<>((Type) field.getData(), new Key(shortName), true);
                     created.setColumnVisibility(field.getColumnVisibility());
-                    fieldMap.put(fieldName, created);
-                    fieldToFieldWithContextMap.put(shortName, fieldName);
+                    String fieldNameWithContext = containsContext ? fieldName : fieldName + ".0";
```

Comment:
Is this making the assumption that if the field data isn't a collection, then the grouping context must be 0?  Is that always true?  Or was this added to account for terms that don't have a grouping context to ensure that they will be processed in the grouping context set in getGroupingInfo?  If so, maybe an identifier other than 0 would be better since 0 is an actual grouping context that could be used.